test/test_browser.py

```diff
@@ -3362,6 +3362,13 @@ def test_async_mainloop(self):
       print(opts)
       self.btest_exit('browser/async_mainloop.cpp', args=['-O' + str(opts), '-sASYNCIFY'])
 
+  @no_firefox('no JSPI support yet')
+  def test_jspi(self):
+    test = test_file('browser/jspi.cpp')
+    self.btest_exit(test, args=['-sASYNCIFY=2',
```

Comment:
The convention I think is just to inline `test_file('browser/jspi.cpp')` here

Reply:
Looks like quite a few have separate files here including the current asyncify tests. I prefer separate as well since my editor behaves better.

Reply:
Sorry I didn't mean inline that source code itself, just that you don't need the `test` var here.

e.g.  `self.btest_exit(test_file('browser/jspi.cpp'), ...`

Reply:
(I agree embedding C++ inside python is not the best idea :)